### PR TITLE
Pre-compile YAStack tree in docker image

### PR DESCRIPTION
**Pull request by fbalicchia**
Hi, please review this PR.
Trying to build the actual image I'm experiencing in the following error
```
E: Unable to locate package linux-headers-4.18.0-20-generic
E: Couldn't find any package by glob 'linux-headers-4.18.0-20-generic'
E: Couldn't find any package by regex 'linux-headers-4.18.0-20-generic'
```

I' ve created a base image with ubuntu 18.10. IMHO can be useful to decouple OS image layer from yastack image

### CODE REVIEW

**Comment by chintan8saaras** [comment]
@fbalicchia this is great ! Sorry, I did not see this commit until now. I'll take a look at it over the weekend. Can I request you to tag me (@chintan8saaras ) next time? That way, I'll not miss it. Appreciate your effort.

File: packaging/Dockerfile.essential [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    libtool \
+    gawk \
+    libc-ares-dev \
+    python \
+    libz-dev \
+    golang-go \
+    curl \
+    && rm -rf /var/lib/apt/lists/* 
+
+RUN update-alternatives --install /usr/bin/gcc gcc /usr/bin/gcc-5 10
```
Comment (chintan8saaras) [inline packaging/Dockerfile.essential]: @fbalicchia What's the idea behind priority 10 here? How does that work? I am not very familiar with update-alternatives

File: packaging/Dockerfile.essential [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    libz-dev \
+    golang-go \
+    curl \
+    && rm -rf /var/lib/apt/lists/* 
+
+RUN update-alternatives --install /usr/bin/gcc gcc /usr/bin/gcc-5 10  
+RUN update-alternatives --install /usr/bin/gcc gcc /usr/bin/gcc-5 20 
+RUN update-alternatives --install /usr/bin/g++ g++ /usr/bin/g++-5 10 
+RUN update-alternatives --install /usr/bin/g++ g++ /usr/bin/g++-5 20 
+RUN update-alternatives --install /usr/bin/cc cc /usr/bin/gcc 30
```
Comment (chintan8saaras) [inline packaging/Dockerfile.essential]: I'm guessing a lower number means a higher priority?

File: packaging/Dockerfile.essential [hunk 3]
```diff
@@ -0,0 +1,52 @@
+ARG DPDK_VERSION=18.05
+ARG OS_VER=18.10
+FROM ubuntu:${OS_VER}
+
+RUN apt-get update && apt-get install -y \
+    g++-5 \
+    gcc-5 \
```
Comment (chintan8saaras) [inline packaging/Dockerfile.essential]: 👍 @fbalicchia Anyway you can generate a PR with this included? I completely missed your changes earlier and committed another docker file.

**Comment by chintan8saaras** [comment]
@fbalicchia Don't quiet get your comment -

```
I' ve created a base image with ubuntu 18.10. IMHO can be useful to decouple OS image layer from yastack image
```

Can you suggest the changes to Dockerfile required to achieve this? Thanks !

File: packaging/Dockerfile.essential (hunk 1 above)
Comment (fbalicchia) [reply packaging/Dockerfile.essential]: Ubuntu 18.10 provied out of the box gcc-7, instead for our case we need gcc-5. In this manner we force OS to use gcc5

**Pull request closed without merging**